File: fabric-interop/android/src/main/java/com/viromedia/bridge/fabric/ViroFabricEvents.java
//  Copyright © 2025 ReactVision. All rights reserved.
//
//  Permission is hereby granted, free of charge, to any person obtaining
//  a copy of this software and associated documentation files (the
//  "Software"), to deal in the Software without restriction, including
//  without limitation the rights to use, copy, modify, merge, publish,
//  distribute, sublicense, and/or sell copies of the Software, and to
//  permit persons to whom the Software is furnished to do so, subject to
//  the following conditions:
//
//  The above copyright notice and this permission notice shall be included
//  in all copies or substantial portions of the Software.
//
//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
//  EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
//  MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
//  IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY
//  CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
//  TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
//  SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.

package com.viromedia.bridge.fabric;

/**
 * Event constants for the Viro Fabric interop layer.
 * These events are compatible with the existing ViroEvents but use the Fabric naming convention.
 */
public class ViroFabricEvents {
    // Core interaction events
    public static final String ON_HOVER = "onHoverViro";
    public static final String ON_CLICK = "onClickViro";
    public static final String ON_TOUCH = "onTouchViro";
    public static final String ON_SWIPE = "onSwipeViro";
    public static final String ON_SCROLL = "onScrollViro";
    public static final String ON_DRAG = "onDragViro";
    public static final String ON_FUSE = "onFuseViro";
    public static final String ON_PINCH = "onPinchViro";
    public static final String ON_ROTATE = "onRotateViro";
    
    // Animation events
    public static final String ON_ANIMATION_START = "onAnimationStartViro";
    public static final String ON_ANIMATION_FINISH = "onAnimationFinishViro";
    
    // Loading events
    public static final String ON_LOAD_START = "onLoadStartViro";
    public static final String ON_LOAD_END = "onLoadEndViro";
    public static final String ON_ERROR = "onErrorViro";
    
    // Video events
    public static final String ON_UPDATE_TIME = "onUpdateTimeViro";
    public static final String ON_BUFFER_START = "onBufferStartViro";
    public static final String ON_BUFFER_END = "onBufferEndViro";
    
    // Physics events
    public static final String ON_COLLIDED = "onCollisionViro";
    public static final String ON_TRANSFORM_DELEGATE = "onNativeTransformDelegateViro";
    
    // AR events
    public static final String ON_TRACKING_UPDATED = "onTrackingUpdatedViro";
    public static final String ON_AMBIENT_LIGHT_UPDATE = "onAmbientLightUpdateViro";
    public static final String ON_ANCHOR_FOUND = "onAnchorFoundViro";
    public static final String ON_ANCHOR_UPDATED = "onAnchorUpdatedViro";
    public static final String ON_ANCHOR_REMOVED = "onAnchorRemovedViro";
    public static final String ON_CAMERA_AR_HIT_TEST_VIRO = "onCameraARHitTestViro";
    public static final String ON_AR_POINT_CLOUD_UPDATE = "onARPointCloudUpdateViro";
    public static final String ON_CAMERA_TRANSFORM_UPDATE = "onCameraTransformUpdateViro";
    
    // Portal events
    public static final String ON_PORTAL_ENTER = "onPortalEnterViro";
    public static final String ON_PORTAL_EXIT = "onPortalExitViro";
    
    // Platform events
    public static final String ON_PLATFORM_UPDATE = "onPlatformUpdateViro";
    public static final String ON_CONTROLLER_STATUS = "onControllerStatusViro";
    public static final String ON_EXIT_VIRO = "onExitViro";
    
    // Container-specific events
    public static final String ON_INITIALIZED = "onInitialized";
    public static final String ON_TRACKING_UPDATED_FABRIC = "onTrackingUpdated";
    public static final String ON_CAMERA_TRANSFORM_UPDATE_FABRIC = "onCameraTransformUpdate";
}

File: ios/ViroFabric/ViroFabricEvents.m
//
//  ViroFabricEvents.m
//  ViroReact
//
//  Created for ReactVision.
//  Copyright © 2025 ReactVision. All rights reserved.
//

#import "ViroFabricEvents.h"

@implementation ViroFabricEvents

// Core interaction events
NSString * const ViroFabricEventOnHover = @"onHoverViro";
NSString * const ViroFabricEventOnClick = @"onClickViro";
NSString * const ViroFabricEventOnTouch = @"onTouchViro";
NSString * const ViroFabricEventOnSwipe = @"onSwipeViro";
NSString * const ViroFabricEventOnScroll = @"onScrollViro";
NSString * const ViroFabricEventOnDrag = @"onDragViro";
NSString * const ViroFabricEventOnFuse = @"onFuseViro";
NSString * const ViroFabricEventOnPinch = @"onPinchViro";
NSString * const ViroFabricEventOnRotate = @"onRotateViro";

// Animation events
NSString * const ViroFabricEventOnAnimationStart = @"onAnimationStartViro";
NSString * const ViroFabricEventOnAnimationFinish = @"onAnimationFinishViro";

// Loading events
NSString * const ViroFabricEventOnLoadStart = @"onLoadStartViro";
NSString * const ViroFabricEventOnLoadEnd = @"onLoadEndViro";
NSString * const ViroFabricEventOnError = @"onErrorViro";

// Video events
NSString * const ViroFabricEventOnUpdateTime = @"onUpdateTimeViro";
NSString * const ViroFabricEventOnBufferStart = @"onBufferStartViro";
NSString * const ViroFabricEventOnBufferEnd = @"onBufferEndViro";

// Physics events
NSString * const ViroFabricEventOnCollided = @"onCollisionViro";
NSString * const ViroFabricEventOnTransformDelegate = @"onNativeTransformDelegateViro";

// AR events
NSString * const ViroFabricEventOnTrackingUpdated = @"onTrackingUpdatedViro";
NSString * const ViroFabricEventOnAmbientLightUpdate = @"onAmbientLightUpdateViro";
NSString * const ViroFabricEventOnAnchorFound = @"onAnchorFoundViro";
NSString * const ViroFabricEventOnAnchorUpdated = @"onAnchorUpdatedViro";
NSString * const ViroFabricEventOnAnchorRemoved = @"onAnchorRemovedViro";
NSString * const ViroFabricEventOnCameraARHitTest = @"onCameraARHitTestViro";
NSString * const ViroFabricEventOnARPointCloudUpdate = @"onARPointCloudUpdateViro";
NSString * const ViroFabricEventOnCameraTransformUpdate = @"onCameraTransformUpdateViro";

// Portal events
NSString * const ViroFabricEventOnPortalEnter = @"onPortalEnterViro";
NSString * const ViroFabricEventOnPortalExit = @"onPortalExitViro";

// Platform events
NSString * const ViroFabricEventOnPlatformUpdate = @"onPlatformUpdateViro";
NSString * const ViroFabricEventOnControllerStatus = @"onControllerStatusViro";
NSString * const ViroFabricEventOnExitViro = @"onExitViro";

// Container-specific events
NSString * const ViroFabricEventOnInitialized = @"onInitialized";
NSString * const ViroFabricEventOnTrackingUpdatedFabric = @"onTrackingUpdated";
NSString * const ViroFabricEventOnCameraTransformUpdateFabric = @"onCameraTransformUpdate";

@end
